src/grpc/infer_handler.h

```diff
@@ -1396,17 +1419,33 @@ InferHandler<
 
     while (cq_->Next(&tag, &ok)) {
       State* state = static_cast<State*>(tag);
+      bool is_notification = false;
       if (state->step_ == Steps::WAITING_NOTIFICATION) {
         State* state_wrapper = state;
         state = state_wrapper->state_ptr_;
         state->context_->SetReceivedNotification(true);
+        is_notification = true;
         LOG_VERBOSE(1) << "Received notification for " << Name() << ", "
                        << state->unique_id_;
+
+        if (state->delay_notification_process_entry_ms_ != 0) {
+          // Will delay the entry to Process by the specified time.
+          // This can be used to test the flow when
+          // 1. cancellation request issued for the request, which invokes
+          // InferResponseComplete callback right before Process.
+          // 2. cancellation request issued for the request during
+          // InferResponseComplete callback right before Process in the
+          // notification thread.
+          LOG_INFO
+              << "Delaying the entry to Process for notification thread by "
+              << state->delay_notification_process_entry_ms_ << " ms...";
+          std::this_thread::sleep_for(std::chrono::milliseconds(
+              state->delay_notification_process_entry_ms_));
+        }
       }
       LOG_VERBOSE(2) << "Grpc::CQ::Next() "
                      << state->context_->DebugString(state);
-      if (!Process(state, ok)) {
-        LOG_VERBOSE(1) << "Done for " << Name() << ", " << state->unique_id_;
```

Comment:
Please, bring back this log

Reply:
Must be I accidentally deleted when clean up the code. I have added it back. Thanks.